src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_FilterModel.cs

```diff
@@ -170,11 +146,23 @@ internal partial class Session
                     }
                     else
                     {
-                        if (filterText.Length <= 1)
+                        // The item didn't match the filter text.  We'll still keep it in the list
+                        // if one of two things is true:
+                        //
+                        //  1. The user has only typed a single character.  In this case they might
+                        //     have just typed the character to get completion.  Filtering out items
+                        //     here is not desirable.
+                        //
+                        //  2. They brough up completion with ctrl-j or through deletion.  In these
+                        //     cases we just always keep all the items in the list.
+
+                        var wasTriggeredByDeleteOrSimpleInvoke =
```

Comment:
But massive thanks for actually explaining in English what the heck this is trying to do.